### PR TITLE
fix(docs): working with plugins - 404, code block language, and more...

### PR DESCRIPTION
**Pull request by muescha**
## Description

changes:

- fix 404 on plugin/package link
- wrong file name
- add code block language (and file name)
- add code block to code value
- explain which API is used
- copy logging example from README to the docs
- remove `]` near link

## Related Issues

- #22258 `docs: add starter for plugins with new plugin recipes` 
- #19267 `Add link checker for Gatsby Docs`

### CODE REVIEW

File: starters/gatsby-starter-plugin/README.md [hunk 1]
```diff
@@ -42,14 +42,14 @@ With `my-gatsby-site` being your Gatsby site, and `my-plugin` being your plugin.
 
 2. Include the plugin in a Gatsby site
 
-Inside of the `gatsby-config` file of your site (in this case, `my-gatsby-site`), include the plugin in the `plugins` array:
+Inside of the `gatsby-config.js` file of your site (in this case, `my-gatsby-site`), include the plugin in the `plugins` array:
 
-```
+```javascript:title=gatsby-config.js
```
Comment (laurieontech) [inline starters/gatsby-starter-plugin/README.md]: The title functionality does not work in READMEs and is only available on .org.

Comment (muescha) [reply starters/gatsby-starter-plugin/README.md]: i assumed that they maybe later also shared the README at the https://gatsbyjs.org/starters/gatsbyjs/gatsby-starter-plugin/ point like other starters (or the https://gatsbyjs.org/starters/gatsbyjs/gatsby-starter-blog/ )

Comment (muescha) [reply starters/gatsby-starter-plugin/README.md]: mhh - i see README is not shared - i will remove the title

Comment (muescha) [reply starters/gatsby-starter-plugin/README.md]: removed the filename

**Comment by muescha** [comment]
@laurieontech removed file name from readme

**Comment by laurieontech** [comment]
Thanks!

**Pull request merged**